pkg/client/entities_client.go

```diff
@@ -14,5 +14,191 @@
+}
+
+func hasSpecificFieldValueInSlice(slice reflect.Value, field string, searchFieldValue string) bool {
+
+	for i := 0; i < slice.Len(); i++ {
+		element := slice.Index(i)
+		if isInvalidReflectionValue(element) {
+			continue
+		}
+
+		idValue := getDynamicFieldFromMapReflection(element, field)
+		if isInvalidReflectionValue(idValue) {
+			continue
+		}
+		if idValue.Interface().(string) == searchFieldValue {
+			return true
+		}
+	}
+
+	return false
+
+}
+
+func capitalizeFirstLetter(str string) string {
+	runes := []rune(str)
+	runes[0] = unicode.ToUpper(runes[0])
+	capitalizedString := string(runes)
+
+	return capitalizedString
+}
+
+func isInvalidReflectionValue(value reflect.Value) bool {
+	if value.Kind() == reflect.Invalid {
+		return true
+	} else {
+		return false
+	}
+}
+
+func genListEntitiesParams(entityType string, entitiesType EntitiesType, ignoreProperties []string) url.Values {
+	params := url.Values{
+		"entitySelector": []string{"type(\"" + entityType + "\")"},
+		"pageSize":       []string{defaultPageSizeEntities},
+		"fields":         []string{getEntitiesTypeFields(entitiesType, ignoreProperties)},
+		"from":           []string{genTimeframeUnixMilliString(defaultEntityDurationTimeframeFrom)},
+		"to":             []string{genTimeframeUnixMilliString(defaultEntityDurationTimeframeTo)},
+	}
+
+	return params
+}
+
+func handleListEntitiesError(entityType string, resp rest.Response, run_extraction bool, ignoreProperties []string, err RespError) (bool, []string, RespError) {
+	if err.WrappedError != nil {
+		retryWithIgnore := false
+		retryWithIgnore, ignoreProperties = validateForPropertyErrors(resp, ignoreProperties, entityType)
+
+		if retryWithIgnore {
+			return run_extraction, ignoreProperties, RespError{}
+		} else {
+			return run_extraction, ignoreProperties, err
+		}
+	} else {
+		return false, ignoreProperties, RespError{}
+	}
+}
+
+type ErrorResponseStruct struct {
+	ErrorResponse ErrorResponse `json:"error"`
+}
+
+type ErrorResponse struct {
+	ErrorCode               int                   `json:"code"`
+	Message                 string                `json:"message"`
+	ConstraintViolationList []ConstraintViolation `json:"constraintViolations"`
+}
+
+type ConstraintViolation struct {
+	Path              string `json:"path"`
+	Message           string `json:"message"`
+	ParameterLocation string `json:"parameterLocation"`
+	Location          string `json:"location"`
+}
+
+func validateForPropertyErrors(resp rest.Response, ignoreProperties []string, entityType string) (bool, []string) {
+	retryWithIgnore := false
+
+	var errorResponse ErrorResponseStruct
+	err2 := json.Unmarshal(resp.Body, &errorResponse)
+
+	if err2 == nil {
+		if errorResponse.ErrorResponse.ErrorCode == 400 {
+			constraintViolationList := errorResponse.ErrorResponse.ConstraintViolationList
+			for _, constraintViolation := range constraintViolationList {
+				if constraintViolation.Path == "fields" {
+					re := regexp.MustCompile(`'([^']+)'.*`)
```

Comment:
you could compile the regex once, outside of the for loop, instead of every time.
possibly even as variable outside the function if this is called a lot

Reply:
additon: would nice to give this a name and comment that tells what it matches. 

See e.g. the ones in internal/regex/regex_patterns.go for name/comment samples. 
Doesn't have to be as extensive as those, as it's only used here though

Reply:
Ok, but it's call a theoretical maximum of 50-100 times, much less in reality

Reply:
ah, ok. I was expecting more with the amount of entities and retries